src/result/error.ts

```diff
@@ -0,0 +1,28 @@
+export const ErrorType = {
+  FeltError: 'FeltError',
+  RelocatableError: 'RelocatableError',
+  Int16Error: 'Int16Error',
+  UintError: 'UintError',
+  MemoryError: 'MemoryError',
+  VMError: 'VMError',
+  InstructionError: 'InstructionError',
+  RunContextError: 'RunContextError',
+} as const;
+
+export class BaseError extends Error {
+  type: string;
+  message: string;
+
+  constructor(type: string, message: string) {
```

Comment:
```suggestion
  constructor(type: typeof ErrorType, message: string) {
```

Reply:
I had to do this `type ErrorTypeKey = (typeof ErrorType)[keyof typeof ErrorType];` becase you are actually passing the value of the ErrorType object.